File: app/src/main/java/com/android/app/utils/PreferenceUtils.java
package com.android.app.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceUtils {
    private static final String PREF_NAME = "app";

    private static final String TOKEN = "token";
    private static final String VERSION = "version";
    private static final String NOTIFICACAO = "notificacao";

    public static SharedPreferences getSPreference(Context context) {
        if (context == null)
            return null;

        SharedPreferences preferences;
        preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);

        return preferences;
    }

    public static void setVersion(Context context, String value) {
        SharedPreferences.Editor editor = getSPreference(context).edit();

        editor.putString(VERSION, value);
        editor.apply();
    }

    public static String getVersion(Context context) {
        return getSPreference(context).getString(VERSION, null);
    }

    public static void setToken(Context context, String value) {
        SharedPreferences.Editor editor = getSPreference(context).edit();

        editor.putString(TOKEN, value);
        editor.apply();
    }

    public static String getToken(Context context) {
        return getSPreference(context).getString(TOKEN, null);
    }

    public static void setNotificacao(Context context, boolean value) {
        SharedPreferences.Editor editor = getSPreference(context).edit();

        editor.putBoolean(NOTIFICACAO, value);
        editor.apply();
    }

    public static boolean getNotificacao(Context context) {
        return getSPreference(context).getBoolean(NOTIFICACAO, false);
    }

}

File: app/src/main/java/com/android/app/utils/PreferenceUtils.kt
package com.android.app.utils

import android.content.Context
import android.content.SharedPreferences
import com.android.app.R

class PreferenceUtils (private var context: Context?) {
    private val PREF_NAME = context!!.getString(R.string.app_name)

    private val AUTH_TOKEN = "auth-token"
    private val USER_NAME = "user_name"
    private val USER_EMAIL = "user_email"
    private val TOKEN_NOTIFICATION = "token_notification"
    private val PUSH_NOTIFICATION = "push_notification"

    private fun getPreference(): SharedPreferences {
        return context!!.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }

    fun getAuthToken(): String? {
        return getPreference().getString(AUTH_TOKEN, null)
    }

    fun setAuthToken(value: String) {
        getPreference().edit().putString(AUTH_TOKEN, value).apply()
    }

    fun getUserName(): String? {
        return getPreference().getString(USER_NAME, null)
    }

    fun setUserName(value: String?) {
        getPreference().edit().putString(USER_NAME, value).apply()
    }

    fun getUserEmail(): String? {
        return getPreference().getString(USER_EMAIL, null)
    }

    fun setUserEmail(value: String?) {
        getPreference().edit().putString(USER_EMAIL, value).apply()
    }

    fun getTokenNotification(): String? {
        return getPreference().getString(TOKEN_NOTIFICATION, null)
    }

    fun setTokenNotification(value: String) {
        getPreference().edit().putString(TOKEN_NOTIFICATION, value).apply()
    }

    fun getPushNotification(): Boolean {
        return getPreference().getBoolean(PUSH_NOTIFICATION, true)
    }

    fun setPushNotification(value: Boolean) {
        getPreference().edit().putBoolean(PUSH_NOTIFICATION, value).apply()
    }

}